### ISSUE TITLE
Can't get changes from junit-3.8.1 to junit-3.8.2.jar

### ISSUE THREAD

**Issue by bvuicik** [siom79/japicmp#146]
The following command runs for more than 15 minutes (not sure it will finish):

`java -jar japicmp-0.9.1-SNAPSHOT-jar-with-dependencies.jar -o /PATH_TO/junit-3.8.1.jar -n /PATH_TO/junit-3.8.2.jar --ignore-missing-classes`

junit-3.8.1.jar and junit-3.8.2.jar were downloaded from https://mvnrepository.com/artifact/junit/junit/.

Running the same command against junit-3.8.jar and junit-3.8.1.jar returns instantly.

I'm using java 1.8.0_101.

**Comment by siom79** [comment]
Thanks for reporting. I could reproduce the issue with the help of your description.

Under rare conditions it could happen that the method `evaluate()` in `CompatibilityChanges` was called recursively.

If you like, you can try the `develop` branch:
```
git clone https://github.com/siom79/japicmp.git
cd japicmp
git checkout develop
mvn install
```

**Comment by siom79** [comment]
Released 0.9.1.
